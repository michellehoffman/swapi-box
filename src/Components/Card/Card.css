@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro');

* {
  margin: 0;
}

.card {
  background-color: rgba(0,0,0,0.5);
  border: 1px solid rgba(255, 232, 31, 1);
  color: rgba(255, 232, 31, 1);
  font-family: 'Source Sans Pro', sans-serif;
  height: 410px;
  margin: 10px;
  min-width: 250px;
  padding: 10px;
  width: 20%;
}

.favorite-button {
  left: 110px;
  position: relative;
  top: -55px;
  width: 30px;
}

.favorite-button,
.residents {
  height: 30px;
}

h2 {
  font-size: 2rem;
  margin: -55px 0 20px 0;
}

.character-detail {
  font-size: 1.45rem;
  margin-left: 20px;
  text-align: left;
  text-transform: uppercase;
}

p,
ul {
  font-size: 1.35rem;
}

p {
  margin: 0 0 15px 40px;
  text-align: left;
}

.residents {
  overflow: scroll;
}

ul {
  list-style-type: none; 
  margin-left: -40px;
}

