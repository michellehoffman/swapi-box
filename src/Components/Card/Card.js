import React from 'react';
import './Card.css';

const Card = ({ type, item, addFavorite }) => {
  const details = { ...item } 
  const { name } = details;

  const peopleDetails = (details) => {
    const { homeworld, species, population } = details;

    return (
      <div>
        <li>Homeworld: { homeworld }</li>
        <li>Species: { species }</li>
        <li>Population: { population }</li>
      </div>
    )
  }

  const planetDetails = (details) => {
    const { terrain, climate, population, residents } = details;

    return (
      <div>
        <li>Terrain: { terrain }</li>
        <li>Climate: { climate }</li>
        <li>Population: { population }</li>
        <li>Residents: { residents }</li>
      </div>
    )
  }

  const vehicleDetails = (details) => {
    const { model, vehicle_class, passengers } = details;

    return (
      <div>
        <li>Model: { model }</li>
        <li>Class: { vehicle_class }</li>
        <li>Number of Passengers: { passengers }</li>
      </div>
    )
  }

  const cardRendered = () => {
    if(type === 'people') {
      return peopleDetails(details);
    } else if(type === 'planets') {
      return planetDetails(details);
    } else if(type === 'vehicles') {
      return vehicleDetails(details);
    }
  }

  const addFavoriteCard = e => {
    addFavorite(item);
    setClass(e);
  }

  const setClass = e => {
    if(e.target.className === 'inactive') {
      e.target.className = 'active';
    } else {
      e.target.className = 'inactive';
    }
  }

  return (
    <div>
      <h2>{ name }</h2>
      <ul>
        { cardRendered() }
      </ul>
      <button className="inactive" onClick={ addFavoriteCard }>+</button>
    </div>
  )
}

export default Card;