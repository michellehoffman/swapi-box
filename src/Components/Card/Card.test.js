import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  let wrapper;

  it('should match snapshot for type: people', () => {
    const person = {
      "name": "Luke Skywalker",
      "homeworld": "https://swapi.co/api/planets/1/",
      "species": [
        "https://swapi.co/api/species/1/"
      ]
    }
    const addFavorite = jest.fn();
    wrapper = shallow(<Card item={ person }
                            type="person"
                            addFavorite={ addFavorite } />)
    
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot for type: planet', () => {
    const planet = {
      "name": "Alderaan",
      "terrain": "grasslands, mountains", 
      "climate": "temperate", 
      "population": "2000000000", 
      "residents": [
        "https://swapi.co/api/people/5/", 
        "https://swapi.co/api/people/68/", 
        "https://swapi.co/api/people/81/"
      ]
    }
    const addFavorite = jest.fn();
    wrapper = shallow(<Card item={ planet }
                            type="planet"
                            addFavorite={ addFavorite } />)
    
    expect(wrapper).toMatchSnapshot();    
  });

  it('should match snapshot for type: vehicle', () => {
    const vehicle = {
      "name": "Sand Crawler", 
      "model": "Digger Crawler", 
      "vehicle_class": "wheeled", 
      "passengers": "30"
    }
    const addFavorite = jest.fn();
    wrapper = shallow(<Card item={ vehicle }
                            type="vehicle"
                            addFavorite={ addFavorite } />)
    
    expect(wrapper).toMatchSnapshot();        
  })
})