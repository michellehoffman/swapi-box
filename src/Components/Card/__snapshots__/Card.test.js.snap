// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot for type: people 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addFavorite={[Function]}
    item={
        Object {
            "homeworld": "https://swapi.co/api/planets/1/",
            "name": "Luke Skywalker",
            "species": Array [
              "https://swapi.co/api/species/1/",
            ],
          }
    }
    type="person"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Luke Skywalker
</h2>,
        <ul />,
        <button
          onClick={[Function]}
>
          +
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Luke Skywalker",
        },
        "ref": null,
        "rendered": "Luke Skywalker",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "+",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Luke Skywalker
</h2>,
          <ul />,
          <button
            onClick={[Function]}
>
            +
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Luke Skywalker",
          },
          "ref": null,
          "rendered": "Luke Skywalker",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "+",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card should match snapshot for type: planet 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addFavorite={[Function]}
    item={
        Object {
            "climate": "temperate",
            "name": "Alderaan",
            "population": "2000000000",
            "residents": Array [
              "https://swapi.co/api/people/5/",
              "https://swapi.co/api/people/68/",
              "https://swapi.co/api/people/81/",
            ],
            "terrain": "grasslands, mountains",
          }
    }
    type="planet"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Alderaan
</h2>,
        <ul />,
        <button
          onClick={[Function]}
>
          +
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Alderaan",
        },
        "ref": null,
        "rendered": "Alderaan",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "+",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Alderaan
</h2>,
          <ul />,
          <button
            onClick={[Function]}
>
            +
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Alderaan",
          },
          "ref": null,
          "rendered": "Alderaan",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "+",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card should match snapshot for type: vehicle 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    addFavorite={[Function]}
    item={
        Object {
            "model": "Digger Crawler",
            "name": "Sand Crawler",
            "passengers": "30",
            "vehicle_class": "wheeled",
          }
    }
    type="vehicle"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Sand Crawler
</h2>,
        <ul />,
        <button
          onClick={[Function]}
>
          +
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sand Crawler",
        },
        "ref": null,
        "rendered": "Sand Crawler",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "+",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Sand Crawler
</h2>,
          <ul />,
          <button
            onClick={[Function]}
>
            +
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sand Crawler",
          },
          "ref": null,
          "rendered": "Sand Crawler",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "+",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
